### ISSUE TITLE
How do you limit the number of columns displayed?

### ISSUE THREAD

**Issue by adiep2501** [altair-viz/altair#1044]
I have a dataset that is heavily skewed to the left. So there are a few columns that are largest and the rest being much smaller. I have around ~500 columns to graph and I can't figure out how to limit them within altair.

I do have a working solution with pandas but it's specific to my use case:
```
alt.Chart(df.nlargest(50, 'count)).mark_bar().encode(
     ...
)
```
This doesn't work well with interactive graphs. For example, if you wanted to see columns with smaller counts, it wouldn't work since it's not there in the first place.

I ultimately want some sort of way to interactive adjust the number of columns you see. Maybe I am looking in the wrong places, but I don't see examples that do that.

**Comment by jakevdp** [comment]
When you say interactive, do you mean scrolling the axes by clicking and dragging, or something different?

**Comment by adiep2501** [comment]
I should have been more clear, sorry about that. I wanted something like a button where you click on the legend and it only shows the columns you are interested in.

For example, lets say we have fruits, stores, and how much every store sold. We could have a graph of all of the stores and their sales, then color the fruits so it's a stacked bar. Some stores don't sell certain fruits, so they could be one color.

I trying to make a graph where if you select a fruit (legend), it only shows the stores that sold at least one of that fruit. That way you don't have to clip the dataset and can see all of it.

**Comment by jakevdp** [comment]
Some discussion here: https://github.com/vega/vega-lite/issues/1657

**Comment by adiep2501** [comment]
I managed to get the effect I wanted with this code:
```
selection = alt.selection_single(empty='all', fields=['fruit'])

base_graph = alt.Chart(compare).add_selection(selection)

bar_graph = base_graph.mark_bar().encode(
    x='store:N',
    y='count:Q'
).transform_filter(
    selection
)

legend_graph = base_graph.mark_point().encode(
    y='fruit'
)

bar_graph | legend_graph
```
I used [this example for the python bits](https://altair-viz.github.io/gallery/select_detail.html?highlight=selection) with the hack you posted in the issue thread.

It's remarkably simple. Although it's going to get ugly when I try to config the text and stuff.

I still have the original problem, how do I limit the number of columns. I can see specific 'fruits' but I can't limit the number of 'stores' I look at. Some 'fruits' are small and viewable but some have hundreds of 'stores'.
